heron/tmaster/src/cpp/manager/stateful-restorer.cpp

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2015 Twitter, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "manager/stateful-restorer.h"
+#include <string>
+#include "manager/stmgrstate.h"
+
+namespace heron {
+namespace tmaster {
+
+StatefulRestorer::StatefulRestorer()
+  : in_progress_(false),
+    restore_txid_(0) {
+}
+
+StatefulRestorer::~StatefulRestorer() { }
+
+void StatefulRestorer::StartRestore(const std::string& _checkpoint_id, const StMgrMap& _stmgrs) {
+  if (in_progress_) {
+    LOG(WARNING) << "Starting a Restore for checkpoint "
```

Comment:
In that case can you include in the log that the restore with id checkpoint_id_in_progress_ will be abandoned?